sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/APISpec.groovy

```diff
@@ -813,4 +833,37 @@ class APISpec extends Specification {
             return Mono.error(new IOException())
         }
     }
+
+    class TransientFailureInjectingHttpPipelinePolicy implements HttpPipelinePolicy {
+
+        private ConcurrentHashMap<String, Boolean> failureTracker = new ConcurrentHashMap<>();
+
+        @Override
+        Mono<HttpResponse> process(HttpPipelineCallContext httpPipelineCallContext, HttpPipelineNextPolicy httpPipelineNextPolicy) {
+            def request = httpPipelineCallContext.httpRequest
+            def key = request.url.toString()
+            if (failureTracker.get(key, false)) {
+                return httpPipelineNextPolicy.process()
+            } else {
+                failureTracker.put(key, true)
+                return request.getBody().flatMap(new Function<ByteBuffer, Publisher<ByteBuffer>>() {
+                    @Override
+                    Publisher<ByteBuffer> apply(ByteBuffer byteBuffer) {
+                        byteBuffer.get()
```

Comment:
Is this just to alter the data to inject the failure? In general, can you leave some notes about the intended behavior of this type?

Reply:
This is to make sure byteBuffers were "touched". I'll leave a comment.

Reply:
done